Automattic/wp-calypso
# FSE: Improve site option block errors

### Summary

This PR is improves the errors associated with the API calls for the site options blocks for full site editing. At this point in time, it addresses the Site Title and Site Description blocks.

* Notices are displayed at the top of the editor, instead of individually per block.
* Notice messages are genersized to be more friendly to the user.
* Displayed block content reflects the error state, instead of being perpetually stuck saying "Site block loading..." (or similar).

<hr>

### Specifics

Removed use of `withNotices`, which seems to be the incorrect method for this case (adding notices to be displayed manually in each block). Instead, notices are added to global `core/notice` store via `createErrorNotice`, which are then handled by the editor and displayed at the top.

New error message wording: 
- "Unable to load site [ title/description ]."
- "Unable to save site [ title/description ]."
- "Error loading [title/description]..."

<hr>

### Screenshots

**Error notice location at top of editor**
<img width="1001" alt="Screen Shot 2019-06-26 at 7 06 02 PM" src="https://user-images.githubusercontent.com/8892849/60221280-168dc300-9847-11e9-8b40-73cc3ee652e9.png">

**Site Title and Site Description block content during an error state**
<img width="450" alt="Screen Shot 2019-06-26 at 7 05 50 PM" src="https://user-images.githubusercontent.com/8892849/60221287-1db4d100-9847-11e9-8492-324971083fe8.png">

<hr>

### Testing

* Run FSE environment via your desired method. 
* Ensure that the Full Site Editing plugin is activated.
* Edit the Header template part.
* Add a Site Title block and a Site Description block. Force an error during the API requests and see the new error notices/wording.

Fixes #34093
